arch/sim/src/sim/up_head.c

```diff
@@ -125,11 +125,6 @@ void up_assert(const char *filename, int line)
 #ifdef CONFIG_BOARD_CRASHDUMP
   board_crashdump(sim_getsp(), this_task(), filename, line);
 #endif
-
-  /* Exit the simulation */
-
-  g_exitcode = EXIT_FAILURE;
-  longjmp(g_simabort, 1);
 }
```

Comment:
This will fail if the assertion occurrs from an interrupt handler or from the IDLE task.  In those cases, the function will return to __assert which will call exit().  Calling exit() from the IDLE thread or from and interrupt handler is fatal.

Reply:
So what would be the proper approach?

Reply:
Condition the longjmp call on the equivalent of the ARM condition:  `if (CURRENT_REGS || (running_task())->flink == NULL)`

The catch is the up_assert() is built into the host OS blob, but those conditions are only available in simulation blob.  I think that up_assert would need to be moved to the simulation blog and some new host_abort() called those conditions are true.

Reply:
I don't think I'm familiar with the sim enough to do those changes correctly. I will take a look later.